modules/components/analytics/core/org.wso2.carbon.pc.analytics.core/src/main/java/org/wso2/carbon/pc/analytics/core/UserLevelMonitor.java

```diff
@@ -0,0 +1,497 @@
+				}
+			}
+		} catch (Exception e) {
+			String errMsg = "PC Analytics core UserLevelMonitoring error.";
+			log.error(errMsg, e);
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Result = " + sortedResult);
+		}
+		return sortedResult;
+	}
+
+	/**
+	 * perform query: SELECT assignUser, COUNT(*) AS completedTotalTasks FROM
+	 *                USER_INVOLVE_SUMMARY_DATA WHERE <date range> GROUP BY assignUser;
+	 * @param filters is used to filter the result
+	 * @return the result as a JSON string
+	 */
+	public String getTotalCompletedTasksVsUserId(String filters) {
+		String sortedResult = "";
+		try {
+			if (Helper.isDASAnalyticsActivated()) {
+				JSONObject filterObj = new JSONObject(filters);
+				long from = filterObj.getLong(AnalyticConstants.START_TIME);
+				long to = filterObj.getLong(AnalyticConstants.END_TIME);
+				String order = filterObj.getString(AnalyticConstants.ORDER);
+				int userCount = filterObj.getInt(AnalyticConstants.NUM_COUNT);
+
+				AggregateField countField = new AggregateField();
+				countField.setFieldName(AnalyticConstants.ALL);
+				countField.setAggregate(AnalyticConstants.COUNT);
+				countField.setAlias(AnalyticConstants.COMPLETED_TOTAL_TASKS);
+
+				ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+				aggregateFields.add(countField);
+
+				AggregateQuery query = new AggregateQuery();
+				query.setTableName(AnalyticConstants.USER_INVOLVE_TABLE);
+				query.setGroupByField(AnalyticConstants.ASSIGN_USER);
+				if (from != 0 && to != 0) {
+					query.setQuery(Helper.getDateRangeQuery(AnalyticConstants.COLUMN_FINISHED_TIME,
+					                                        from, to));
+				}
+				query.setAggregateFields(aggregateFields);
+
+				if (log.isDebugEnabled()) {
+					log.debug(Helper.getJSONString(query));
+				}
+
+				String result = AnalyticsRestClient
+								.post(Helper.getURL(AnalyticConstants.ANALYTICS_AGGREGATE),
+						        Helper.getJSONString(query));
+
+				JSONArray unsortedResultArray = new JSONArray(result);
+				Hashtable<String, Integer> table = new Hashtable<>();
+
+				if (unsortedResultArray.length() != 0) {
+					for (int i = 0; i < unsortedResultArray.length(); i++) {
+						JSONObject jsonObj = unsortedResultArray.getJSONObject(i);
+						JSONObject values = jsonObj.getJSONObject("values");
+						String assignee = values.getJSONArray("assignUser").getString(0);
+						int totalInvolvedTime = values.getInt("completedTotalTasks");
+						table.put(assignee, totalInvolvedTime);
+					}
+					sortedResult = Helper.getIntegerValueSortedList(table, "assignUser",
+					                                                "completedTotalTasks", order,
+					                                                userCount);
+				}
+			}
+		} catch (Exception e) {
+			String errMsg = "PC Analytics core UserLevelMonitoring error.";
+			log.error(errMsg, e);
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Result = " + sortedResult);
+		}
```

Comment:
Debug log is too abstract.. Please add descriptive one..  Check other places as well.